py3status/modules/window_manipulation.py

```diff
@@ -0,0 +1,396 @@
+                'border': w.border,
+                'is_focused': w.focused,
+                'is_urgent': w.urgent,
+                'is_floating': getattr(w, 'floating', False),  # python2
+                'is_scratchpad': is_scratchpad,
+            })
+
+        return new_scratchpad, new_window
+
+    def _manipulate_and_refresh(self, new_scratchpad=None, new_window=None):
+        """
+        For lack of a better word, we're trying to manipulate the data to make
+        the results more usable for the users. This is usually done by creating
+        new data and/or alter existing data. We strives to let data move through
+        without any or very little manipulation.
+        """
+        self.shared_data = new_window
+        count_floating = count_scratchpad = count_window = \
+            count_urgent = count_urgent_sp = 0
+
+        if new_scratchpad:
+            for s in new_scratchpad:
+                if s['is_urgent']:
+                    count_urgent_sp += 1
+
+        new_data = []
+        for w in new_window:
+            count_window += 1
+            # send all values (for focused_window) out to the format (as placeholders)
+            # so the users can do some simple things like printing name only.
+            if w['is_focused']:
+                self.id = w.get('id')
+                self.window_name = w.get('name')
+                self.class_name = w.get('class')
+            if w['is_scratchpad']:
+                count_scratchpad += 1
+            if w['is_urgent']:
+                count_urgent += 1
+
+            # when scrolling, we want to ignore this error because we already
+            # fixed it on update. we turned this into a boolean too.
+            try:
+                if 'on' in w['is_floating']:  # counts exclusively on updating
+                    w['is_floating'] = True
+                    count_floating += 1
+                else:
+                    w['is_floating'] = False
+            except:
+                if w['is_floating']:  # counts exclusively on scrolling
+                    count_floating += 1
+
+            new_data.append(self.py3.safe_format(self.format_window, w))
+
+        # send all counts out to the format (as placeholders)
+        self.count_floating = count_floating
+        self.count_scratchpad = count_scratchpad
+        self.count_urgent = count_urgent
+        self.count_urgent_sp = count_urgent_sp
+        self.count_window = count_window
+
+        # put everything together and do a refresh
+        format_separator = self.py3.safe_format(self.format_separator)
+        self.fmt_window = self.py3.composite_join(format_separator, new_data)
+        self.py3.update()
+
+    def _switch_window_manipulation(self, direction):
+        """
+        Users can scroll windows with mouse, click to focus. If an urgent occurs,
+        we will use a different click to focus. Disabled by default.
+        """
+        windows, f = self.shared_data, 'is_focused'
```

Comment:
Quick answer no.

To be honest I'm still not sure what your doing here with the urgent stuff or more importantly why.  I haven't actually looked in detail as the code has other issues imho